File: TestDagger2/src/main/java/someday/com/testdagger2/mvp/di/component/AppComponent.java
package someday.com.testdagger2.mvp.di.component;

import javax.inject.Singleton;

import dagger.Component;
import someday.com.testdagger2.mvp.di.module.AppModule;
import someday.com.testdagger2.mvp.di.module.HttpModule;

/**
 * Author: 海晨忆
 * Date: 2018/2/23
 * Desc:
 */
@Singleton
@Component(modules = {AppModule.class, HttpModule.class})
public interface AppComponent
{


}

File: TestDagger2KtMvp/src/main/java/someday/com/testdagger2kt/nomvp/di/AppComponent.kt
package someday.com.testdagger2kt.nomvp.di

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import someday.com.testdagger2kt.nomvp.MyApplication

@Singleton2
@Component(modules = arrayOf(
        AndroidSupportInjectionModule::class,
        ActivityBuilderModule::class,
        ViewModelModule::class))
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): AppComponent.Builder

        fun build(): AppComponent
    }

    fun inject(instance: MyApplication)
}